wpep0008.md

```diff
@@ -0,0 +1,50 @@
+| Key | Value |
+| ---: | :--- |
+| **Title** | Wazo integration tests framework |
+| **Number** | 0008 |
+| **Date** | 2023-08-28 |
+| **Authors** | Jesse Sopel <jesse.sopel@wazo.io>, Sebastien Duthil <sduthil@wazo.io> |
+| **Status** | draft |
+
+## Abstract
+
+Wazo integration tests: Explicit the direction of evolution of integration tests
```

Comment:
```suggestion
Make explicit the direction of evolution of integration tests towards a more pytest-oriented paradigm.
```

Reply:
Make explicit the desire to migrate towards a more pytest-oriented paradigm in integration tests. ?

Reply:
For example, with the TestCase format:

```python
# helpers/base.py
class BaseAssetLaunchingTestCase(AssetLaunchingTestCase):

    assets_root = os.path.join(os.path.dirname(__file__), '../..', 'assets')
    asset = 'base'
    service = 'agid'

    @classmethod
    def make_agid(cls):
        port = cls.service_port(4573, 'agid')
        return AgidClient('127.0.0.1', port)

class IntegrationTest(unittest.TestCase):
    asset_cls = BaseAssetLaunchingTestCase
    agid: AgidClient

    @classmethod
    def setUpClass(cls):
        cls.agid = cls.asset_cls.make_agid()

# conftest.py
from .helpers import base as asset

def pytest_collection_modifyitems(session, config, items):
    item.parent.own_markers[0].args[0])

@pytest.fixture(scope='session')
def base():
    asset.BaseAssetLaunchingTestCase.setUpClass()
    try:
        yield
    finally:
        asset.BaseAssetLaunchingTestCase.tearDownClass()

# test_handlers.py
@use_asset('base')
class TestHandlers(IntegrationTest):
    def test_monitoring(self):
        recv_vars, recv_cmds = self.agid.monitoring()
        assert recv_cmds['Status'] == 'OK'
```

Could be converted to:
```python
# helpers/base.py
class BaseAssetLaunchingHelper(AbstractAssetLaunchingHelper):
    assets_root = Path(__file__).parents[2] / 'assets'
    asset = 'base'
    service = 'agid'

    @cached_class_property
    def agid(cls) -> AgidClient:
        port = cls.service_port(4573, 'agid')
        return AgidClient('127.0.0.1', port)

# conftest.py
@pytest.fixture(scope='session')
def base_asset() -> Generator[BaseAssetLaunchingHelper, None, None]:
    yield from make_asset_fixture(BaseAssetLaunchingHelper)

# test_handlers.py
def test_monitoring(base_asset: BaseAssetLaunchingHelper) -> None:
    recv_vars, recv_cmds = base_asset.agid.monitoring()
    assert recv_cmds['Status'] == 'OK'
```